### ISSUE TITLE
Broken on old version of Ruby

### ISSUE THREAD

**Issue by isqad** [rspec-given/rspec-given#31]
Hello! 
After release v3.8.1 I started to getting errors:
```
NoMethodError: undefined method `source_location' for #<Binding:0x0000000a05b360>
  /bundle/2.2/gems/given_core-3.8.1/lib/given/module_methods.rb:35:in `location_of'
  /bundle/2.2/gems/given_core-3.8.1/lib/given/extensions.rb:254:in `Then'
```
on Ruby v2.3

Why minor version and can You constraint ruby version in gemspec?

Regards, Andrew.

**Comment by searls** [comment]
Whoops, sorry about this. The change was to remove an `eval` call in favor of `source_location`, which was added in 1.9 but must not be available in whatever context it was being called from in this case.

Will add this to my backlog but may not have time soon for it. Would love a PR that patched this by either branching on Ruby versions (basically changing the last release's change to being conditional on `RUBY_VERSION`) or otherwise digging into the binding to find the right thing to ask for the `source_location` of

**Comment by timcraft** [comment]
> The change was to remove an eval call in favor of source_location, which was added in 1.9 but must not be available in whatever context it was being called from in this case.

Binding#source_location wasn't added until Ruby 2.6:
https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/

So the change in ad82bcd effectively changes the ruby version required by this gem.

@searls Could you please make the change backwards compatible like [this change to pry](https://github.com/pry/pry/pull/1904), or alternatively bump the required ruby version of the gem to 2.6.0 (if that was the intent). Something like this should be backwards compatible:

```ruby
def self.location_of(block)
  if block.binding.respond_to?(:source_location)
    block.binding.source_location
  else
    eval "[__FILE__, __LINE__]", block.binding
  end
end
```

**Comment by searls** [comment]
Thanks for pointing this out -- fixed in 3.8.2
